lib/vector-buffers/src/buffer_usage_data.rs

```
@@ -15,6 +15,25 @@ use crate::{
     spawn_named,
 };
 
+fn u64_to_i64(value: u64) -> i64 {
+    // If we ever have to deal with more than 9,223,372,036,854,775,808 events or bytes _in a single
+    // update_, this conversion will break counters. That number is effectively impossibly large,
```

This comment basically describes what `try_from` does below. Alternatively, let's remove it and handle the Result below. By emitting an `error!` (instead of silently defaulting to `i64::MAX`) we might catch some rare bug.

---

Note that this is also called in `CategoryMetrics::{in,de}crement` as well. This will need different handling.

By that token, however, we should also not `clamp`, but that was kinda the point of the original work.

Not sure how to proceed here.

---

I was thinking of the following:

```rust
let result = match i64::try_from(value) {
    Ok(v) => v,
    Err(e) => {
        error!("Defaulting to i64::MAX. Failed to convert value to i64: {e}");
        i64::MAX
    }
};
```

---

Would you like to see the same pattern replace the `clamp` as well?

---

Nice, I like this version better.